riverdriver/riversqlite/internal/dbsqlc/river_leader.sql

```diff
@@ -0,0 +1,76 @@
+CREATE TABLE river_leader (
+    elected_at timestamp NOT NULL,
+    expires_at timestamp NOT NULL,
+    leader_id text NOT NULL,
+    name text PRIMARY KEY NOT NULL DEFAULT 'default' CHECK (name = 'default'),
+    CONSTRAINT name_length CHECK (length(name) > 0 AND length(name) < 128),
+    CONSTRAINT leader_id_length CHECK (length(leader_id) > 0 AND length(leader_id) < 128)
+);
+
+-- name: LeaderAttemptElect :execrows
+INSERT INTO /* TEMPLATE: schema */river_leader (
+    leader_id,
+    elected_at,
+    expires_at
+) VALUES (
+    @leader_id,
+    coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec')),
+    datetime(coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec')), 'subsec', cast(@ttl as text))
+)
+ON CONFLICT (name)
+    DO NOTHING;
+
+-- name: LeaderAttemptReelect :execrows
+INSERT INTO /* TEMPLATE: schema */river_leader (
+    leader_id,
+    elected_at,
+    expires_at
+) VALUES (
+    @leader_id,
+    coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec')),
+    datetime(coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec')), 'subsec', cast(@ttl as text))
+)
+ON CONFLICT (name)
+    DO UPDATE SET
+        expires_at = EXCLUDED.expires_at
+    WHERE
+        leader_id = EXCLUDED.leader_id;
+
+-- name: LeaderDeleteExpired :execrows
+DELETE FROM /* TEMPLATE: schema */river_leader
+WHERE expires_at < coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec'));
+
+-- name: LeaderGetElectedLeader :one
+SELECT *
+FROM /* TEMPLATE: schema */river_leader;
+
+-- name: LeaderInsert :one
+INSERT INTO /* TEMPLATE: schema */river_leader(
+    elected_at,
+    expires_at,
+    leader_id
+) VALUES (
+    coalesce(cast(sqlc.narg('elected_at') AS text), cast(sqlc.narg('now') AS text), datetime('now', 'subsec')),
+    coalesce(cast(sqlc.narg('expires_at') AS text), datetime(coalesce(cast(sqlc.narg('now') AS text), datetime('now', 'subsec')), 'subsec', cast(@ttl as text))),
+    @leader_id
+) RETURNING *;
+
+-- name: LeaderResign :execrows
+DELETE
+FROM /* TEMPLATE: schema */river_leader
+WHERE leader_id = @leader_id;
+
+-- WITH currently_held_leaders AS (
+--   SELECT *
+--   FROM /* TEMPLATE: schema */river_leader
+--   WHERE leader_id = @leader_id::text
+--   FOR UPDATE
+-- ),
+-- notified_resignations AS (
+--     SELECT pg_notify(
+--         concat(coalesce(sqlc.narg('schema')::text, current_schema()), '.', @leadership_topic::text),
+--         json_build_object('leader_id', leader_id, 'action', 'resigned')::text
+--     )
+--     FROM currently_held_leaders
+-- )
+-- DELETE FROM /* TEMPLATE: schema */river_leader USING notified_resignations;
```

Comment:
want to leave a comment on this?

Reply:
Ah actually, that was just scratch I was using as a reference while writing the new query. I just removed it. Thx.